routes/transaction.go

```diff
@@ -2548,6 +2547,287 @@ func (fes *APIServer) TransferDAOCoin(ww http.ResponseWriter, req *http.Request)
+		)
+		return
+	}
+
+	if requestData.SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername == "" &&
+		requestData.BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername == "" {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("CreateOrCancelDAOCoinLimitOrder: Must provide either a "+
+				"BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername or SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername "+
+				"or both"),
+		)
+		return
+	}
+
+	if !validateExchangeRateCoinsToSellPerCoinToBuyParam(requestData) {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("CreateOrCancelDAOCoinLimitOrder: Must provide only one of the "+
+				"ScaledExchangeRateCoinsToSellPerCoinToBuy or ExchangeRateCoinsToSellPerCoinToBuy params"),
+		)
+		return
+	}
+
+	if !validateQuantityToBuyParam(requestData) {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("CreateOrCancelDAOCoinLimitOrder: Must provide only one of the "+
+				"ScaledExchangeRateCoinsToSellPerCoinToBuy or ExchangeRateCoinsToSellPerCoinToBuy params"),
+		)
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CreateOrCancelDAOCoinLimitOrder: Problem fetching utxoView: %v", err),
+		)
+		return
+	}
+
+	// Decode the transactor public key
+	transactorPublicKeyBytes, _, err := fes.GetPubKeAndProfileEntryForUsernameOrPublicKeyBase58Check(
+		requestData.TransactorPublicKeyBase58Check,
+		utxoView,
+	)
+	if err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CreateOrCancelDAOCoinLimitOrder: Problem decoding transactor public key %s: %v",
+				requestData.TransactorPublicKeyBase58Check, err))
+		return
+	}
+
+	// Compute the additional transaction fees as specified by the request body and the node-level fees.
+	additionalOutputs, err := fes.getTransactionFee(
+		lib.TxnTypeDAOCoinTransfer,
+		transactorPublicKeyBytes,
+		requestData.TransactionFees,
+	)
+	if err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CreateOrCancelDAOCoinLimitOrder: TransactionFees specified in Request body are invalid: %v", err),
+		)
+		return
+	}
+
+	buyingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
+	sellingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
+
+	if requestData.BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername != "" {
+		buyingCoinPublicKey, _, err = fes.GetPubKeAndProfileEntryForUsernameOrPublicKeyBase58Check(
+			requestData.BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername,
+			utxoView,
+		)
+		if err != nil {
+			return
+		}
+	}
+
+	if requestData.SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername != "" {
+		sellingCoinPublicKey, _, err = fes.GetPubKeAndProfileEntryForUsernameOrPublicKeyBase58Check(
+			requestData.SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername,
+			utxoView,
+		)
+		if err != nil {
+			return
+		}
+	}
+
+	scaledExchangeRateCoinsToSellPerCoinToBuy := requestData.ScaledExchangeRateCoinsToSellPerCoinToBuy
+	if requestData.ExchangeRateCoinsToSellPerCoinToBuy != 0 {
```

Comment:
```suggestion
	if requestData.ExchangeRateCoinsToSellPerCoinToBuy != float64(0) {
```

Reply:
Not an issue here. The compiler is smart enough to know it's a float64 type when we use the 0 literal. The code reads a bit cleaner as-is